src/core/collector/mod.rs

```diff
@@ -203,6 +212,24 @@ impl Collector {
 		})
 	}
 
+	pub async fn map_subxt_event(&mut self, event: &SubxtEvent) -> color_eyre::Result<Vec<SubxtEvent>> {
```

Comment:
The naming is confusing a bit wrt return type. A doc comment describing how it works would be helpful.

Looking at the code, it seems to parse a `SubxtEvent` and produce more `SubxtEvents`. This sounds a bit odd. :) I think we need to split them up, such that block events are no longer mixed with `subxt` events and this function actually decodes all block events, which makes sense.

Reply:
I agree. I wanted to change it but hesitated to use an event with a single value.